pkg/network/kubemacpool_test.go

```diff
@@ -0,0 +1,190 @@
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %s", errorList[0].Error())
+				Expect(errorList[0].Error()).To(Equal("failed to set RangeEnd: invalid mac address. Multicast addressing is not supported. Unicast addressing must be used. The first octet is 0X3"))
+			})
+		})
+
+		Context("When the mac address is valid and multicast bit is off", func() {
+			It("should NOT return an error", func() {
+				clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+				errorList := validateKubeMacPool(clusterConfig)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+	})
+
+	Describe("Testing fillDefaultsKubeMacPool", func() {
+		Context("When kubeMacPool is nil", func() {
+			It("should NOT return an error and do nothing", func() {
+				currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+				previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+					KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+				errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+				Expect(currentClusterConfig.KubeMacPool).To(BeNil())
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+		Context("When the user hasn't explicitly requested a range", func() {
+			Context("When a previous kubeMacPool exits", func() {
+				It("should use the previous kubeMacPoll range for the current one, and not return an error", func() {
+					previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "02:00:00:00:00:00", RangeEnd: "0A:FF:FF:FF:FF:FF"}}
+					currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "", RangeEnd: ""}}
+					errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+					Expect(errorList).To(BeEmpty())
+					Expect(currentClusterConfig.KubeMacPool.RangeStart).To(Equal(previousClusterConfig.KubeMacPool.RangeStart))
+					Expect(currentClusterConfig.KubeMacPool.RangeEnd).To(Equal(previousClusterConfig.KubeMacPool.RangeEnd))
+
+				})
+			})
+
+			Context("When a previous kubeMacpool doesn't exits", func() {
+				It("should generate a new range for the current kubeMacPoll, and not return an error", func() {
+					previousClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{}
+					currentClusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+						KubeMacPool: &opv1alpha1.KubeMacPool{RangeStart: "", RangeEnd: ""}}
+					errorList := fillDefaultsKubeMacPool(currentClusterConfig, previousClusterConfig)
+					Expect(errorList).To(BeEmpty())
+					Expect(currentClusterConfig.KubeMacPool.RangeStart).To(Not(Equal("")), "RangeStart:")
```

Comment:
That message in case of an error "RangeStart:" is not saying anything. Could you explain what went unexpected instead?